### ISSUE TITLE
--overlap option with two adapters

### ISSUE THREAD

**Issue by agileta** [marcelm/cutadapt#153]
I am attempting to use cutadapt on GBS data in order to simultaneously trim for a barcode (ranges from 4-8bps in length) anchored at the beginning of the read, an Illumina adapter on the 3' end, and quality trim below Q20. The issue is that I would like to set a minimum overlap of 6 or 8bp for the 3' adapter (seems like the default of 3 is very low), but if I use the -O option with a number greater than the length of the barcode (4bp in the case below), cutadapt crashes. Is there any way to combine these commands and use the overlap parameter, or is this something that would need to be done in two separate commands then?

./bin/cutadapt -g ^ACTA -a AGATCGGAAGAGC -O 8 -q 20 ./fastqs/80500.fq.gz -o ./80500.Q20CutA.fastq.gz -e 0.20 2> ./report_80500.Q20CutA.txt

**Comment by marcelm** [comment]
I’ve fixed the problem now that cutadapt crashes when the given minimum overlap is longer than the adapter length.

The default minimum overlap is quite low, but it was chosen carefully. You can calculate that you lose only 0.07 bases per read on average through random matches if you set it to 3; it’s not really necessary to go higher, but that may also depend on the application (so feel free to set it higher).

Note that by default, cutadapt trims *at most one* adapter per read. It searches each of the ones provided and trims the best matching one. If you want to trim two adapters, then you need to add the option `-n 2`, which will search for an adapter twice, see also [the section in the docs](http://cutadapt.readthedocs.org/en/stable/guide.html#trimming-more-than-one-adapter-from-each-read). I’ll improve the support for adapter pairs/linked adapters soon.

Are you comfortable with getting the fixed sources from git? If not, I can prepare a new release (which I was planning to do anyway).

**Comment by agileta** [comment]
Yes, I can get the fixed sources. Appreciate the assistance and the clarification on adapter pair trimming!

**Comment by m-ocejo** [comment]
Hi Martin,
I am having issues in trimming my primers using cutadapt.
After merging paired-end reads, I wanted to trim both forward and reverse primers (both degraded). The forward primers are usually linked to its 5´ with 5 "N" (random bases). To make it more clear, my reads are like this:
NNNNNforwardprimerMYSEQUENCEreverseprimer

after running the command, it seemed to be working with the forward, but not with the reverse. I also found that sometimes the linker bases could be 4 or 6. 
Thanks in advance!

**Comment by marcelm** [comment]
Could you provide an example sequence where you do not get the result that you would expect?

**Comment by m-ocejo** [comment]
I used this command after merging:

cutadapt -g NNNNNGTGCCAGCMGCCGCGGTAA -a ATTAGAWACCCBDGTAGTCC -e 0.1 -n 2 --trimmed-only -f fastq -o cutadapt/Sample00001.fastq paired/Sample00001.fastq

I guess that I have figured it out. I added the option "-n 2" and it seemed that i worked OK. I am having too many unique sequences (singletons) on my analysis; more or less 73%. I was wondering if it might be due to cutadapt. Is there any possibility to create a file in which all the discarded sequences are stored?

Thank you!
